src/terminal/char_grid/cursor.rs

```diff
@@ -24,10 +24,20 @@ pub struct Cursor {
     pub coords: Coords,
     pub style: Styles,
     pub text_style: Styles,
+    config: Config,
```

Comment:
> This will raise lifetime issues. Ultimately Cursor and CharGrid and so on are all owned Terminal, which also owns the Config. You can't store a reference into the same object.

Okay, this makes sense, I didn't actually think it would work as phrased...I am still trying to understand the various options available here as far as Rust is concerned so I hope you don't mind if I bikeshed a little. Specifically, I was thinking that rather than copying the `Config` around everywhere or having it owned by `Terminal` and referenced elsewhere as `&Config` that we could do something like allocate new `Config`s on the heap using either `Rc<Config>` or `Arc<Config>`--if I understand correctly, the choice depends on whether we need the `Config` to be available in more than one context.

My goal isn't just to avoid boiler plate but also to only pass references as a way to discourage copying and potentially storing configuration somewhere that would become a liability--if we have a UI menu somewhere that makes changes to a Config object, that change should propagate by virtue of that fact that we are changing a reference rather than explicitly calling an update method everywhere it is needed.

> Not sure if you know this, but when you call a method like terminal.move_cursor, Rust will insert as many dereference operators before terminal as it needs to do get to a type that has a move_cursor method.

I didn't realize this, although I knew such a thing was possible...I think you pointed it out at one point in one of my earlier PRs. So I guess if it's not possible/practical/desirable to give each struct a `Rc<Config>`, what do you think about storing the `Config` on the `CharGrid` instead and relying on the `Terminal` to deref to `CharGrid` to access it? Does that even make sense?